sky/clouds/aws.py

```diff
@@ -399,16 +401,31 @@ def make_deploy_resources_variables(self,
 
         user_security_group = skypilot_config.get_nested(
             ('aws', 'security_group_name'), None)
-        if resources.ports is not None:
-            # Already checked in Resources._try_validate_ports
-            assert user_security_group is None
-            security_group = USER_PORTS_SECURITY_GROUP_NAME.format(
-                cluster_name_on_cloud)
-        elif user_security_group is not None:
-            assert resources.ports is None
-            security_group = user_security_group
-        else:
+        if user_security_group is not None and not isinstance(
+                user_security_group, str):
+            for profile in user_security_group:
+                if fnmatch.fnmatchcase(cluster_name.name_on_cloud,
+                                       list(profile.keys())[0]):
+                    user_security_group = list(profile.values())[0]
+                    break
+        security_group = user_security_group
+        if security_group is None:
             security_group = DEFAULT_SECURITY_GROUP_NAME
+            if resources.ports is not None:
+                # Already checked in Resources._try_validate_ports
+                security_group = USER_PORTS_SECURITY_GROUP_NAME.format(
+                    cluster_name.name_on_cloud)
+        elif security_group is not None and resources.ports is not None:
```

Comment:
Seems this is ensured by the previous if statement.
```suggestion
        elif resources.ports is not None:
```